catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -697,13 +713,21 @@ public MetacardImpl createMetacard(SolrDocument doc) throws MetacardCreationExce
       return Collections.emptyList();
     }
 
+    boolean isNrtCommit = false;
     List<SolrInputDocument> docs = new ArrayList<>();
     for (Metacard metacard : metacards) {
       docs.add(getSolrInputDocument(metacard));
+      if (commitNrtMetacardType.contains(metacard.getMetacardType().getName())) {
+        isNrtCommit = true;
```

Comment:
❓ Does that mean that if only one metacard out of all specified requires an Nrt commit that all of them will be done that way? Should we split the metacards into 2 sets?

Reply:
I think once there is one `commitWithin` sent to Solr, all pending documents will be soft committed.  Unless Solr has a way to do a soft commit per document and not per collection.